lib/rubocop/cop/performance/hash_each_methods.rb

```diff
@@ -55,17 +67,32 @@ def autocorrect(node)
           caller, first_method = *receiver
 
           lambda do |corrector|
-            if receiver.method?(:hash)
-              correct_hash(node, corrector)
-            else
+            if first_method.eql?(:keys) || first_method.eql?(:values)
+              return correct_implicit(node, corrector) if receiver
+                                                          .child_nodes
+                                                          .first.nil?
+
               new_source = caller.source + ".each_#{first_method[0..-2]}"
               corrector.replace(node.loc.expression, new_source)
+            else
+              return correct_implicit(node, corrector) if receiver.nil?
```

Comment:
Maybe this could be changed to an `elsif` + `else`?

```ruby
if ...
  ...
elsif !receiver
  correct_implicit(node, corrector)
else
  correct_plain_each(node, corrector)
end
```